subprojects/core/src/testFixtures/groovy/org/gradle/internal/operations/BuildOperationExecutorSupport.java

```diff
@@ -90,19 +90,34 @@ public BuildOperationExecutor build() {
                 ? this.executorFactory
                 : new DefaultExecutorFactory();
 
-            BuildOperationRunner buildOperationRunner = new DefaultBuildOperationRunner(
-                CurrentBuildOperationRef.instance(),
-                timeSupplier,
-                new DefaultBuildOperationIdFactory(),
-                executionListenerFactory
-            );
-
             return new DefaultBuildOperationExecutor(
-                buildOperationRunner,
+                buildRunner(),
                 CurrentBuildOperationRef.instance(),
                 queueFactory,
                 executorFactory,
                 parallelismConfiguration);
         }
+
+        private BuildOperationRunner buildRunner() {
+            BuildOperationRunner runner;
+            if (this.runner == null) {
+                BuildOperationTimeSupplier timeSupplier = this.timeSupplier != null
+                    ? this.timeSupplier
+                    : System::currentTimeMillis;
+                DefaultBuildOperationRunner.BuildOperationExecutionListenerFactory executionListenerFactory = this.executionListenerFactory != null
+                    ? this.executionListenerFactory
+                    : () -> DefaultBuildOperationRunner.BuildOperationExecutionListener.NO_OP;
+
+                runner = new DefaultBuildOperationRunner(
```

Comment:
💅 Can we just do `return new DefaultBuildOperationRunner()` here and avoid local `BuildOperationRunner runner`?